x-pack/plugin/mapper-version/src/main/java/org/elasticsearch/xpack/versionfield/VersionEncoder.java

```diff
@@ -35,7 +35,7 @@
  *  lexically in ASCII sort order. Numeric identifiers always have lower precedence than non-numeric identifiers.
  * </ul>
  */
-class VersionEncoder {
+public class VersionEncoder {
```

Comment:
Good point, actually these changes are a leftover from the previous attempts to support Version without impacting on the Search/Painless implementation, but since we went with a more complete solution (see https://github.com/elastic/elasticsearch/pull/85990), we don't need them anymore.